src/services/CertificateGenerationService.ts

```diff
@@ -133,63 +121,139 @@ class CertificateGenerationService {
+                        ExpiryDate: (testType.testExpiryDate) ? moment(testType.testExpiryDate).format("DD.MM.YYYY") : undefined,
+                        EarliestDateOfTheNextTest: (
+                          (testResult.vehicleType === VEHICLE_TYPES.HGV || testResult.vehicleType === VEHICLE_TYPES.TRL)
+                          && (testResult.testTypes.testResult === TEST_RESULTS.PASS || testResult.testTypes.testResult === TEST_RESULTS.PRS)
+                        ) ?
+                            moment(testType.testAnniversaryDate).subtract(2, "months").add(1, "days").format("DD.MM.YYYY") :
+                            moment(testType.testAnniversaryDate).format("DD.MM.YYYY"),
+                        SeatBeltTested: (testType.seatbeltInstallationCheckDate) ? "Yes" : "No",
+                        SeatBeltPreviousCheckDate: (testType.lastSeatbeltInstallationCheckDate) ? moment(testType.lastSeatbeltInstallationCheckDate).format("DD.MM.YYYY") : "\u00A0",
+                        SeatBeltNumber: testType.numberOfSeatbeltsFitted,
+                        ...defects
+                    };
+            case "RWT_DATA":
+                const weightDetails = await this.getWeightDetails(testResult);
+                let defectRWTList: any;
+                if (testResult.testTypes.testResult === TEST_RESULTS.FAIL) {
+                                defectRWTList = [];
+                                testResult.testTypes.defects.forEach((defect: any) => {
+                                       defectRWTList.push(this.formatDefect(defect));
+                                   });
+                } else {
+                    defectRWTList = undefined;
+                }
+
+                const resultPass: IRoadworthinessCertificateData = {
+                    Dgvw: weightDetails.dgvw,
+                    Weight2: weightDetails.weight2,
+                    VehicleNumber:  testResult.vehicleType === VEHICLE_TYPES.TRL ? testResult.trailerId : testResult.vrm,
+                    Vin:  testResult.vin,
+                    IssuersName: testResult.testerName,
+                    DateOfInspection: moment(testType.testTypeStartTimestamp).format("DD.MM.YYYY"),
+                    TestStationPNumber: testResult.testStationPNumber,
+                    DocumentNumber: testType.certificateNumber,
+                    Date: moment(testType.testTypeStartTimestamp).format("DD.MM.YYYY"),
+                    Month: moment(testType.testTypeStartTimestamp).format("MMMM"),
+                    Year: moment(testType.testTypeStartTimestamp).year(),
+                    Defects: defectRWTList,
+                    IsTrailer: testResult.vehicleType === VEHICLE_TYPES.TRL
+                };
+                return resultPass;
+        }
     }
 
+    /**
+     * Retrieves the vehicle weight details for Roadworthisness certificates
+     * @param testResult
+     */
+    public async getWeightDetails(testResult: any) {
```

Comment:
Look over missing test cases - leave it up to you what is important
![image](https://user-images.githubusercontent.com/50110006/70245827-a7f64f00-176e-11ea-88b8-1467b8fb3c24.png)